core/src/test/scala/integration/kafka/api/AdminClientTest.scala

```diff
@@ -58,11 +63,148 @@ class AdminClientTest extends IntegrationTestHarness with Logging {
+    client.purgeDataBefore(Map((tp, PurgeRequest.HIGH_WATERMARK))).get()
+    consumer.seekToBeginning(Collections.singletonList(tp))
+    assertEquals(10L, consumer.position(tp))
+  }
+
+  @Test
+  def testConsumeAfterPurge() {
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    assertEquals(10, consumer.poll(1000).count())
+
+    client.purgeDataBefore(Map((tp, 3L))).get()
+    consumer.seek(tp, 1)
+    assertEquals(7, consumer.poll(1000).count())
+
+    client.purgeDataBefore(Map((tp, 8L))).get()
+    consumer.seek(tp, 1)
+    assertEquals(2, consumer.poll(1000).count())
+  }
+
+  @Test
+  def testLogStartOffsetAfterPurge() {
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    client.purgeDataBefore(Map((tp, 3L))).get()
+
+    for (i <- 0 until serverCount)
+      assertEquals(3, servers(i).replicaManager.getReplica(tp).get.logStartOffset)
+  }
+
+  @Test
+  def testLogStartOffsetAfterAsyncPurge() {
+    val consumer = consumers.head
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    client.purgeDataBefore(Map((tp, 3L)))
+    Thread.sleep(1000)
+
+    for (i <- 0 until serverCount)
+      assertEquals(3, servers(i).replicaManager.getReplica(tp).get.logStartOffset)
+  }
+
+  @Test
+  def testPurgeDataBeforeWithOffsetsForTimes() {
+    val consumer = consumers.head
+
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+    assertEquals(0L, consumer.offsetsForTimes(Map(tp -> new JLong(0L)).asJava).get(tp).offset())
+
+    client.purgeDataBefore(Map((tp, 5L))).get()
+    assertEquals(5L, consumer.offsetsForTimes(Map(tp -> new JLong(0L)).asJava).get(tp).offset())
+
+    client.purgeDataBefore(Map((tp, PurgeRequest.HIGH_WATERMARK))).get()
+    assertNull(consumer.offsetsForTimes(Map(tp -> new JLong(0L)).asJava).get(tp))
+  }
+
+  @Test
+  def testPurgeDataBeforeWithException() {
+    val consumer = consumers.head
+
+    consumer.subscribe(Collections.singletonList(topic))
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(0)
+      !consumer.assignment.isEmpty
+    }, "Expected non-empty assignment")
+
+    sendRecords(producers.head, 10, tp)
+
+    killRandomBroker()
+    // wait for metadata update
+    Thread.sleep(2000)
```

Comment:
This test with offline brokers is needed when we define LW to min offset of all replicas. But it is not needed anymore after we change it to beh min offset of all live replicas. I simply removed this part.